app/controllers/api/v2/rh_cloud/inventory_controller.rb

```diff
@@ -28,6 +28,31 @@ def generate_report
           }, status: :ok
         end
 
+        api :GET, "/organizations/:organization_id/rh_cloud/missing_hosts", N_("Grab hosts that are missing in RH Cloud")
+        param :organization_id, Integer, required: true, desc: N_("Set the current organization context for the request")
+        def get_hosts
+          organization_id = params[:organization_id]
+          payload = InsightsMissingHosts.where(organization_id: organization_id)
+
+          render :json => payload
+        end
+
+        api :POST, "/organizations/:organization_id/rh_cloud/missing_hosts", N_("Grab hosts that are missing in RH Cloud")
+        param :organization_id, Integer, required: true, desc: N_("Set the current organization context for the request")
+        param :search_term, String, required: true, desc: N_("Scoped search string for host removal")
+        def remove_hosts
+          organization_id = params[:organization_id]
+          search_term = params[:search_term]
+
+          task = ForemanTasks.sync_task(ForemanInventoryUpload::Async::RemoveInsightsHostsJob, search_term, organization_id)
```

Comment:
```suggestion
          task = ForemanTasks.async_task(ForemanInventoryUpload::Async::RemoveInsightsHostsJob, search_term, organization_id)
```
We don't want the thread to wait for the action to finish.